### PR TITLE
Add firefox/sync (#9490, Fixes #9519, #9516)

### LINKED ISSUE #9519: Firefox Sync Coding
## Description

In https://github.com/mozilla/bedrock/issues/9516 we build a Sync page. This issue tracks coding.

- Url: /firefox/sync/
- Open for translation

---

#### Success Criteria
- [ ] Page is coded
- [ ] Redirect to /accounts/ is removed

### PR DESCRIPTION
**Pull request by amychurchwell**
## Description
Adds firefox/sync

## Issue / Bugzilla link
#9490
#9519
#9516

## Testing
[Copy](https://docs.google.com/document/d/1LE3SwAOvSajQ_3TC_lt87tj2Fxokxhe55BFiCngVuBA/edit)
[Design](https://github.com/mozilla/bedrock/issues/9518#issuecomment-706292804)

### CODE REVIEW

File: bedrock/firefox/templates/firefox/browsers/quantum.html [hunk 1]
```diff
@@ -15,20 +15,19 @@
...
-  include_highres_image=True,
-  include_cta=True,
-  heading_level=1
-) %}
-<p>{{ ftl('firefox-quantum-was') }}</p>
-{{ download_firefox_thanks() }}
-<p><a class="mzp-c-cta-link" href="https://blog.mozilla.org/blog/2017/11/14/introducing-firefox-quantum/?utm_source=mozilla.org-firefox-browsers-quantum&utm_medium=referral&utm_campaign=seo-quantum&amp;utm_content=hero&amp;entrypoint=mozilla.org-firefox-browsers-quantum">{{ ftl('learn-more-about') }}</a></p>
-{% endcall %}
-
 <div class="mzp-l-content">
```
Comment (amychurchwell) [inline bedrock/firefox/templates/firefox/browsers/quantum.html:18]: unrelated to this PR, but I felt like the hero for the recent firefox/browsers/quantum page should be inside the `mzp-l-content` 🤷‍♀️

**Comment by amychurchwell** [comment]
Adding Do Not Merge until demo approved by Raphael & Chad

**Comment by amychurchwell** [comment]
Approved!

File: bedrock/firefox/templates/firefox/browsers/quantum.html (hunk 1 above)
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/browsers/quantum.html:18]: The hero component already has an `mzp-l-content` container within it, so nesting it inside another one here creates some odd looking cropped content at smaller screen widths

Perhaps using one of the non-default [container theme classes](https://protocol.mozilla.org/patterns/templates/content-container.html), such as `mzp-t-content-xl` might look closer to how you want?

image

File: l10n/en/firefox/sync.ftl [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# This Source Code Form is subject to the terms of the { -brand-name-mozilla } Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/firefox/sync/
+
+firefox-sync = { -brand-name-firefox-sync }
```
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:7]: Fluent IDs should use a standard prefix that associates them with the page where they are being used. This helps avoid naming collisions. See: https://bedrock.readthedocs.io/en/latest/l10n.html#ftl-string-ids

File: bedrock/firefox/templates/firefox/sync.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+{% block page_title %}{{ ftl('firefox-sync') }}{% endblock %}
+
+{% block page_desc %}{{ ftl('access-and-sync') }}{% endblock %}
+
+{% block page_css %}
+ {{ css_bundle('sync') }}
+{% endblock %}
+
+{% block content %}
+<div class="mzp-l-content">
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:18]: Same as for /quantum, the hero component should not be nested within another `mzp-l-content`

File: bedrock/firefox/templates/firefox/sync.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  {% call hero(
+    title=ftl('sync-your-firefox'),
+    class='mzp-has-image mzp-t-firefox mzp-t-product-firefox',
+    image_url='img/firefox/accounts/trailhead/value-knowledge.jpg',
+    include_highres_image=True,
+    include_cta=True,
+    heading_level=1
+  ) %}
+  <p>{{ ftl('access-and-sync') }}</p>
+  {{ download_firefox_thanks() }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:28]: It feels a bit weird that this page is all about Sync, yet the primary CTA is still "Download Firefox" when I view the page in Firefox? At first i didn't notice the Sync instructions further down the page. Would it be worth replacing this with a sign-in button for Firefox users?

File: bedrock/firefox/templates/firefox/sync.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    ) %}
+
+      <ol class="mzp-u-list-styled">
+        <li>{{ ftl('open-the-menu') }}</li>
+        <li>{{ ftl('click-get-started') }}</li>
+        <li>{{ ftl('enter-an-email') }}</li>
+        <li>{{ ftl('click-next-to') }}</li>
+      </ol>
+      <p>{{ ftl('after-you-check') }}</p>
+      <a class="mzp-c-cta-link" href="https://support.mozilla.org/en-US/kb/firefox-accounts-managing-account-data">{{ ftl('learn-more') }}</a>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:65]: Should this link have some utms for tracking?

File: l10n/en/firefox/sync.ftl [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+access-and-sync = Access and sync your bookmarks, passwords, open tabs and more — everywhere you use { -brand-name-firefox }.
+sync-your-firefox = { -brand-name-sync } your { -brand-name-firefox } on any device
+privacy-made-easy = Privacy made easy
+all-you-need = All you need for { -brand-name-firefox-sync } is a { -brand-name-firefox-account }.
+encrypt-your-data = Encrypt your data
+your-encryption-key = Your encryption key remains a secret, only known to you and your authorized devices.
+feel-safe = Feel safe
+we-store-your = We store your encrypted data on our servers so we can't read it.
```
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:15]: Nit: curly `’`
```suggestion
we-store-your = We store your encrypted data on our servers so we can’t read it.
```

File: l10n/en/firefox/sync.ftl [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+start-syncing-in = Start syncing in four easy steps:
+
+open-the-menu = Open the menu in the top right of { -brand-name-firefox } and select <strong>“Sign in to { -brand-name-sync }</strong>.”
+click-get-started = Click <strong>“Get started”</strong> in the tab that opens.
+enter-an-email = Enter an email address and password to <strong>“Create a { -brand-name-firefox-account }.”</strong>
+click-next-to = Click <strong>“Next”</strong> to get a verification sent to your email.
+after-you-check = After you check your email and click the verification link. { -brand-name-firefox } will start automatically syncing in the background.
+
+# https://support.mozilla.org/en-US/kb/firefox-accounts-managing-account-data
```
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:24]: Remove /en-US/ from link

File: bedrock/firefox/redirects.py [hunk 8]
```diff
@@ -310,9 +310,6 @@ def firefox_channel(*args, **kwargs):
     # Bug 1239863, 1329931
     redirect(r'^firefox/os(/.*)?$', 'https://support.mozilla.org/products/firefox-os'),
 
-    # Bug 1252332
-    redirect(r'^sync/?$', 'firefox.accounts'),
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py:314]: This is now a 404 - did you mean to redirect it to /firefox/sync/?

**Review by alexgibson — APPROVED** [review]
A few nits / suggestions but overall the page is looking good 👍

**Review by alexgibson — CHANGES_REQUESTED** [review]
Oops, meant to request changes instead of approve. Trying again

File: l10n/en/firefox/sync.ftl (hunk 2 above)
Comment (alexgibson) [reply l10n/en/firefox/sync.ftl:7]: I also realize now that we missed this with the /quantum page :(

File: bedrock/firefox/templates/firefox/sync.html (hunk 4 above)
Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/sync.html:29]: Agreed! Done.

File: l10n/en/firefox/sync.ftl (hunk 2 above)
Comment (amychurchwell) [reply l10n/en/firefox/sync.ftl]: PR to update firefox/quantum: https://github.com/mozilla/bedrock/pull/9623/files
thanks Alex :)

File: bedrock/firefox/redirects.py [hunk 9]
```diff
@@ -311,7 +311,7 @@ def firefox_channel(*args, **kwargs):
     redirect(r'^firefox/os(/.*)?$', 'https://support.mozilla.org/products/firefox-os'),
 
     # Bug 1252332
-    redirect(r'^sync/?$', 'firefox.accounts'),
+    redirect(r'^sync/?$', '/firefox/sync/'),
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py:314]:
```suggestion
    redirect(r'^sync/?$', 'firefox.sync'),
```

File: bedrock/firefox/templates/firefox/sync.html [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    ) %}
+
+      <ol class="mzp-u-list-styled">
+        <li>{{ ftl('firefox-sync-open-the-menu') }}</li>
+        <li>{{ ftl('firefox-sync-click-get-started') }}</li>
+        <li>{{ ftl('firefox-sync-enter-an-email') }}</li>
+        <li>{{ ftl('firefox-sync-click-next-to') }}</li>
+      </ol>
+      <p>{{ ftl('firefox-sync-after-you-check') }}</p>
+      <a class="mzp-c-cta-link" href="https://support.mozilla.org/en-US/kb/firefox-accounts-managing-account-data?utm_source=www.mozilla.org-firefox-sync&amp;utm_medium=referral&amp;utm_campaign=seo" rel="external noopener" data-cta-text="More Help Managing your Firefox Account" data-cta-type="link">{{ ftl('firefox-sync-learn-more') }}</a>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:76]: Remove `/en-US/`

File: l10n/en/firefox/sync.ftl [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/firefox/sync/
+
+firefox-sync-firefox-sync = { -brand-name-firefox-sync }
+
+firefox-sync-access-and-sync = Access and sync your bookmarks, passwords, open tabs and more — everywhere you use { -brand-name-firefox }.
+
+firefox-sync-sync-your-firefox = { -brand-name-sync } your { -brand-name-firefox } on any device
```
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:11]: Should this be a verb instead brand name?
```suggestion
firefox-sync-sync-your-firefox = Sync your { -brand-name-firefox } on any device
```

File: l10n/en/firefox/sync.ftl [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+firefox-sync-click-get-started = Click <strong>“Get started”</strong> in the tab that opens.
+enter-an-email = Enter an email address and password to <strong>“Create a { -brand-name-firefox-account }.”</strong>
+
+firefox-sync-click-next-to = Click <strong>“Next”</strong> to get a verification sent to your email.
+
+firefox-sync-after-you-check = After you check your email and click the verification link. { -brand-name-firefox } will start automatically syncing in the background.
+
+# https://support.mozilla.org/kb/firefox-accounts-managing-account-data
+firefox-sync-learn-more = More help managing your Firefox Account
```
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:36]:
```suggestion
firefox-sync-learn-more = More help managing your { -brand-name-firefox-account }
```

File: bedrock/firefox/templates/firefox/sync.html [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+  image_url='img/firefox/accounts/trailhead/value-knowledge.jpg',
+  include_highres_image=True,
+  include_cta=True,
+  heading_level=1
+  ) %}
+  <p>{{ ftl('firefox-sync-access-and-sync') }}</p>
+
+  {{ download_firefox_thanks() }}
+  <p class="fxa-cta">
+    {{ fxa_button(
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:31]: I think you still need to include the JS dependencies for the FxA button? https://bedrock.readthedocs.io/en/latest/firefox-accounts.html#cta-button-dependencies

**Review by alexgibson — CHANGES_REQUESTED** [review]
Looking good, just a few small fixups and the FxA button needs the JS libs for flow params.

**Review by alexgibson — APPROVED** [review]

**Pull request merged**